jax/interpreters/xla.py

```diff
@@ -601,15 +601,18 @@ def _xla_callable_device(nreps, backend, device, arg_devices):
     else:
       assert False  # Unreachable given the error check in _xla_callable
 
-def _xla_callable_args(c, avals, tuple_args):
+def _xla_callable_args(c, avals, tuple_args, replicated=None):
   if not tuple_args:
-    xla_args = [xb.parameter(c, i, aval_to_xla_shape(a))
+    if replicated is None:
```

Comment:
Shouldn't this be outside of the `if not tuple_args`?

Reply:
No, since in the tuple_args case we _do_ want to pass a single `None` as the replication argument. This all is driven by the fact that XLA uses an empty `std::vector` on both tuple parameters and non-tuple parameters to signal "no particular replication annotation"; we're modeling that as `None`.